### DISCUSSION TITLE
MQTT Username and password

### DISCUSSION THREAD

**Discussion by CreativeImpulse** [tsightler/ring-mqtt#754] [General Discussion]
Hi,

I'm sure this is a noddy question but I don't see it in the documentation.  I have a working MQTT server, happily serving away for z2m on my docker HA installation. It has a username and password set. I just installed the Ring integration, it happily took my Ring details with no issues. However, I noted that MQTT server username and password are not mentioned in the installation process. I have seen reference to the use of the mqtt.js library for this scenario in completely unrelated applications and that involved the use of "username: 'usr', password: 'pswd'" in the mqtt_options, so having tracked down the code and looked at it, it seemed to match and I tried:

"mqtt_options": "username: 'blah', password: 'wibble'",

in the config.json. I issued as docker container restart and not a lot happened... I had expected the ring mqtt instances to pop up in the HA system. What am I missing? Javascript is not really my language, so this could be really trivial.

Cheers

**Comment by tsightler** [comment]
It's the first config option discussed in the documentation (mqtt_url):

https://github.com/tsightler/ring-mqtt/wiki/Configuration-Details#global-configuration-options

**Comment by CreativeImpulse** [comment]
Thankyou, somehow I missed that.

 

It still isn’t working but I think that may be topic related, I need to get a listener onto the mqtt server and make sure the messages are getting through. I note my current MQTT listener on my HA for z2m has a topic set as the base level for it’s communications, base_topic: zigbee2mqtt. Looks like the ring integration drops it’s messages into a topic “ring”, which won’t match. I’m assuming that’s why I can’t see anything just now. So I’m going to have to work out how to rework that without changing the existing setup if possible.

 

Thanks for the help.

**Reply by tsightler** [reply]
I mean it's fully expected that different tools use different base topics, I have many different plugins which all use different base topics, so I have no idea why that would be an issue unless you are using specific ACLs that block topics.

**Comment by CreativeImpulse** [comment]
Hi,  

 

Yeah I think, I still have an issue with the MQTT connection. I got a promiscuous listener going on the HA this evening and there was nothing from Ring. I think I need to go in search of some log files and see if we have some sort of refused connection. It looks like the z2m is configured to use a base topic, the HA just takes anything it sees.

 

Cheers

 

Rob
